packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -121,7 +121,7 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
                 }
                 if (event.action === nsfw.actions.RENAMED) {
                     this.pushDeleted(watcherId, this.resolvePath(event.directory, event.oldFile!));
-                    this.pushAdded(watcherId, this.resolvePath(event.directory, event.newFile!));
+                    this.pushAdded(watcherId, this.resolvePath(event.newDirectory!, event.newFile!));
```

Comment:
Should not we fall back to `directory` if `newDirectory` is missing? It would be good to know when `newDirectory` is present or not, i.e. probably move within the same directory, does not change directory.

Reply:
As far as I understand [the `RENAMED` event is defined by `nsfw` as always exposing a `newDirectory` property](https://github.com/Axosoft/nsfw/blob/84bf5e48eabfa6414eec783a09ec12d753ecb4c9/index.d.ts#L34).

I tested renaming files in the same directory on Linux, Windows and Mac and the `newDirectory` seems to be always present.

I certainly can't be sure it's always the case, so if you prefer me to add a fallback on `event.directory` I can certainly add that.

Reply:
let's fallback, since [mentioned docs](https://github.com/Axosoft/nsfw/blob/84bf5e48eabfa6414eec783a09ec12d753ecb4c9/index.d.ts#L39) says that it's only useful on linux

Reply:
I'm not really sure about the meaning of that comment. In any case, I've added the fallback.